File: 11.盛最多水的容器.py
#
# @lc app=leetcode.cn id=11 lang=python3
#
# [11] 盛最多水的容器
#

# @lc code=start
# 双指针法
class Solution:
    def maxArea(self, height: List[int]) -> int:
        res = 0
        i = 0
        j = len(height) - 1
        while i < j:
            area = (j - i) * min(height[i], height[j])
            res = max(area, res)
            if height[i] < height[j]:
                i += 1
            else:
                j -= 1
        return res
# @lc code=end

File: 数据结构/数组/11.盛最多水的容器.java
/*
 * @lc app=leetcode.cn id=11 lang=java
 *
 * [11] 盛最多水的容器
 */

// @lc code=start
class Solution {
    public int maxArea(int[] height) {
        int max = 0;
        for(int i = 0, j = height.length - 1; i < j ; ){
            int min_height = height[i] < height[j] ? height[i++] : height[j--];
            int area = (j - i + 1) * min_height;
            max = Math.max(max, area);
        }
        return max;
    }
}
// @lc code=end

